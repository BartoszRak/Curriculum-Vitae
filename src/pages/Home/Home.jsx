import React from 'react'
import classNames from 'classnames'
import uuidv4 from 'uuid/v4'

import Projects from './Projects'
import Bio from './Bio'
import Skills from './Skills'
import Workflow from './Workflow'
import Hobbies from './Hobbies'

import useStyle from './Home.style'

const contentList = [Projects, Bio, Skills, Workflow, Hobbies].map(obj => ({
  component: obj,
  id: uuidv4(),
}))

export function Home() {
  const classes = useStyle()
  return (
    <main className={classes.root} role="main">
      {contentList.map((item, index) => {
        const { component: Component, id } = item
        return (
          <Component
            classes={{
              section: classNames(classes.holder, {
                [classes.fairHolder]: index % 2 !== 0,
                [classes.backgroundHolder]: index % 2 === 0,
              }),
            }}
            key={id}
          />
        )
      })}
    </main>
  )
}

export default Home
