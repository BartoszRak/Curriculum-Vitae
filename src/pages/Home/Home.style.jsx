export default ({ palette, spacing }) => ({
  bioHolder: {
    background: palette.secondary.main,
  },
  divider: {
    background: palette.secondary.main,
    height: spacing.unit * 2.5,
  },
  holder: {
    padding: '50px 15%',
  },
  projectsHolder: {
    backgroundAttachment: 'fixed',
  },
  root: {
    alignItems: 'stretch',
    display: 'flex',
    flexDirection: 'column',
  },
})
