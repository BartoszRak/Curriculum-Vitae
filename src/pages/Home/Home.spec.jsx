import React from 'react'
import { shallow } from 'enzyme'

import { Home } from './Home'

jest.mock('./ProjectsList', () => 'ProjectsListMock')

const initComponent = overrides => {
  const mockProps = {
    classes: {},
  }
  const mockMethods = {}
  const wrapper = shallow(<Home {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('global: Home', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
