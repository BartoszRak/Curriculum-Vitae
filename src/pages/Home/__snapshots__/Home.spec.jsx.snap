// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Home should render as expected 1`] = `
<div>
  <div
    className=""
    style={
      Object {
        "backgroundImage": "url(landscape-mountains-nature-dark-blur.jpg)",
      }
    }
  >
    <ProjectsMock />
  </div>
</div>
`;
